Data races when running unit tests

**Describe the bug**
When running unit tests, the go race checker found three instances of data race.

**Logs**
```
	github.com/kuadrant/authorino		coverage: 0.0% of statements
	github.com/kuadrant/authorino/api/v1beta2		coverage: 0.0% of statements
	github.com/kuadrant/authorino/api/v1beta3		coverage: 0.0% of statements
ok  	github.com/kuadrant/authorino/controllers	(cached)	coverage: 46.4% of statements
	github.com/kuadrant/authorino/controllers/builder		coverage: 0.0% of statements
	github.com/kuadrant/authorino/controllers/builder/mocks		coverage: 0.0% of statements
	github.com/kuadrant/authorino/controllers/mocks		coverage: 0.0% of statements
	github.com/kuadrant/authorino/controllers/mocks/controller-runtime/client		coverage: 0.0% of statements
ok  	github.com/kuadrant/authorino/pkg/auth	(cached)	coverage: 77.2% of statements
	github.com/kuadrant/authorino/pkg/auth/mocks		coverage: 0.0% of statements
	github.com/kuadrant/authorino/pkg/context		coverage: 0.0% of statements
ok  	github.com/kuadrant/authorino/pkg/evaluators	(cached)	coverage: 32.4% of statements
--- FAIL: TestOPAExternalUrlWithTTL (4.00s)
    testing.go:1617: race detected during execution of test
FAIL
coverage: 76.5% of statements
FAIL	github.com/kuadrant/authorino/pkg/evaluators/authorization	4.129s
--- FAIL: TestOIDCProviderVerifierRefresh (4.01s)
    testing.go:1617: race detected during execution of test
FAIL
coverage: 70.8% of statements
FAIL	github.com/kuadrant/authorino/pkg/evaluators/identity	10.411s
	github.com/kuadrant/authorino/pkg/evaluators/identity/mocks		coverage: 0.0% of statements
ok  	github.com/kuadrant/authorino/pkg/evaluators/metadata	(cached)	coverage: 77.9% of statements
ok  	github.com/kuadrant/authorino/pkg/evaluators/response	(cached)	coverage: 69.4% of statements
?   	github.com/kuadrant/authorino/pkg/expressions	[no test files]
ok  	github.com/kuadrant/authorino/pkg/expressions/cel	(cached)	coverage: 78.7% of statements
ok  	github.com/kuadrant/authorino/pkg/health	(cached)	coverage: 100.0% of statements
	github.com/kuadrant/authorino/pkg/httptest		coverage: 0.0% of statements
ok  	github.com/kuadrant/authorino/pkg/index	(cached)	coverage: 77.0% of statements
	github.com/kuadrant/authorino/pkg/index/mocks		coverage: 0.0% of statements
ok  	github.com/kuadrant/authorino/pkg/json	(cached)	coverage: 81.7% of statements
ok  	github.com/kuadrant/authorino/pkg/jsonexp	(cached)	coverage: 46.8% of statements
ok  	github.com/kuadrant/authorino/pkg/log	(cached)	coverage: 42.3% of statements
ok  	github.com/kuadrant/authorino/pkg/metrics	(cached)	coverage: 96.1% of statements
	github.com/kuadrant/authorino/pkg/metrics/mocks		coverage: 0.0% of statements
ok  	github.com/kuadrant/authorino/pkg/oauth2	(cached)	coverage: 87.5% of statements
ok  	github.com/kuadrant/authorino/pkg/service	(cached)	coverage: 77.2% of statements
	github.com/kuadrant/authorino/pkg/trace		coverage: 0.0% of statements
ok  	github.com/kuadrant/authorino/pkg/utils	(cached)	coverage: 75.0% of statements
--- FAIL: TestStartWorker (3.00s)
    testing.go:1617: race detected during execution of test
FAIL
coverage: 80.0% of statements
FAIL	github.com/kuadrant/authorino/pkg/workers	6.016s
	github.com/kuadrant/authorino/pkg/workers/mocks		coverage: 0.0% of statements
FAIL
```

**Race checker logs**
Not in order

[race1.log](https://github.com/user-attachments/files/24643492/race1.log)
[race2.log](https://github.com/user-attachments/files/24643490/race2.log)
[race3.log](https://github.com/user-attachments/files/24643491/race3.log)

**Environment (please complete the following information):**
- Run on commit https://github.com/Kuadrant/authorino/commit/ddab19c9e61f384830ff1d52ef94f0a6299b463a
- go version go1.25.5 X:nodwarf5 linux/amd64

**Additional context**
Related to #555

--- a/pkg/evaluators/authorization/opa.go
+++ b/pkg/evaluators/authorization/opa.go
@@ -116,6 +116,13 @@ func (opa *OPA) Clean(_ context.Context) error {
 	return opa.ExternalSource.cleanupRefresher()
 }
 
+// GetRego returns the current Rego policy in a thread-safe manner
+func (opa *OPA) GetRego() string {
+	opa.mu.RLock()
+	defer opa.mu.RUnlock()
+	return opa.Rego
+}
+
 func (opa *OPA) updateRego(rego string, ctx context.Context, force bool) (bool, error) {
 	opa.mu.Lock()
 	defer opa.mu.Unlock()

--- a/pkg/evaluators/authorization/opa_test.go
+++ b/pkg/evaluators/authorization/opa_test.go
@@ -153,11 +153,11 @@ func TestOPAExternalUrlWithTTL(t *testing.T) {
 	}(opa)
 
 	assert.NilError(t, err)
-	assert.Check(t, strings.Contains(opa.Rego, "GET"))
+	assert.Check(t, strings.Contains(opa.GetRego(), "GET"))
 	assert.Check(t, opa.ExternalSource.refresher != nil)
 
 	time.Sleep(4 * time.Second)
-	assert.Check(t, strings.Contains(opa.Rego, "POST"))
+	assert.Check(t, strings.Contains(opa.GetRego(), "POST"))
 }
 
 func TestOPAClean(t *testing.T) {

--- a/pkg/evaluators/identity/jwt.go
+++ b/pkg/evaluators/identity/jwt.go
@@ -155,6 +155,13 @@ func (v *oidcProviderVerifier) Clean(ctx gocontext.Context) error {
 	return v.refresher.Stop()
 }
 
+// GetProvider returns the current OIDC provider in a thread-safe manner
+func (v *oidcProviderVerifier) GetProvider() *oidc.Provider {
+	v.mu.RLock()
+	defer v.mu.RUnlock()
+	return v.provider
+}
+
 func (v *oidcProviderVerifier) getOpenIdProvider(ctx gocontext.Context, force bool) *oidc.Provider {
 	v.mu.Lock()
 	defer v.mu.Unlock()

--- a/pkg/evaluators/identity/jwt_test.go
+++ b/pkg/evaluators/identity/jwt_test.go
@@ -3,6 +3,7 @@ package identity
 import (
 	"context"
 	"fmt"
+	"sync"
 	"testing"
 	"time"
 
@@ -126,11 +127,15 @@ func TestOIDCProviderVerifierInternalError(t *testing.T) {
 }
 
 func TestOIDCProviderVerifierRefresh(t *testing.T) {
+	var mu sync.Mutex
 	count := 0
 	authServer := httptest.NewHttpServerMock(oidcServerHost, map[string]httptest.HttpServerMockResponseFunc{
 		"/.well-known/openid-configuration": func() httptest.HttpServerMockResponse {
+			mu.Lock()
 			count += 1
-			return oidcServerMockResponse(count)
+			currentCount := count
+			mu.Unlock()
+			return oidcServerMockResponse(currentCount)
 		},
 	})
 	defer authServer.Close()
@@ -149,9 +154,13 @@ func TestOIDCProviderVerifierRefresh(t *testing.T) {
 	assert.Check(t, verifier.refresher != nil)
 
 	time.Sleep(4 * time.Second)
-	assert.Equal(t, 2, count)
+	mu.Lock()
+	currentCount := count
+	mu.Unlock()
+	assert.Equal(t, 2, currentCount)
 	verifier, _ = jwtVerifier.(*oidcProviderVerifier)
-	assert.Equal(t, fmt.Sprintf("http://%v/auth?count=2", oidcServerHost), verifier.provider.Endpoint().AuthURL)
+	provider := verifier.GetProvider()
+	assert.Equal(t, fmt.Sprintf("http://%v/auth?count=2", oidcServerHost), provider.Endpoint().AuthURL)
 }
 
 func TestOIDCProviderVerifierRefreshDisabled(t *testing.T) {

--- a/pkg/workers/worker_test.go
+++ b/pkg/workers/worker_test.go
@@ -2,6 +2,7 @@ package workers
 
 import (
 	"context"
+	"sync"
 	"testing"
 	"time"
 
@@ -9,35 +10,53 @@ import (
 )
 
 func TestStartWorker(t *testing.T) {
+	var mu sync.Mutex
 	val := 0
 	worker, err := StartWorker(context.TODO(), 2, func() {
+		mu.Lock()
 		val += 1
+		mu.Unlock()
 	})
 	defer func(worker Worker) {
 		_ = worker.Stop()
 	}(worker)
 	assert.NilError(t, err)
 
-	assert.Equal(t, val, 0)
+	mu.Lock()
+	currentVal := val
+	mu.Unlock()
+	assert.Equal(t, currentVal, 0)
 
 	time.Sleep(3 * time.Second)
 
-	assert.Equal(t, val, 1)
+	mu.Lock()
+	currentVal = val
+	mu.Unlock()
+	assert.Equal(t, currentVal, 1)
 }
 
 func TestStopWorker(t *testing.T) {
+	var mu sync.Mutex
 	val := 0
 	worker, err := StartWorker(context.TODO(), 2, func() {
+		mu.Lock()
 		val += 1
+		mu.Unlock()
 	})
 	assert.NilError(t, err)
 
-	assert.Equal(t, val, 0)
+	mu.Lock()
+	currentVal := val
+	mu.Unlock()
+	assert.Equal(t, currentVal, 0)
 
 	err = worker.Stop()
 	assert.NilError(t, err)
 
 	time.Sleep(3 * time.Second)
 
-	assert.Equal(t, val, 0)
+	mu.Lock()
+	currentVal = val
+	mu.Unlock()
+	assert.Equal(t, currentVal, 0)
 }

